[stable/rabbitmq-ha] Fixed invalid value

austinorth opened the pull request:
#### What this PR does / why we need it:
Sets the `service.clusterIP` variable to an empty value and commented out, following the standard set in other charts for optional empty values, to avoid invalid value error. Also updated readme and bumped chart version.

Signed-off-by: Austin Orth <aorth@niche.com>

#### Which issue this PR fixes
  - fixes #18463

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

**CC:** @steven-sheehy

Commit:
[stable/rabbitmq-ha] Fixed invalid value

Set the service.clusterIP variable to an empty value and commented out,
following the standard set in other charts for optional empty values, to
avoid invalid value error. Also updated readme and bumped chart version.

Signed-off-by: Austin Orth <aorth@niche.com>

Gsantomaggio commented on stable/rabbitmq-ha/values.yaml:
```diff
@@ -315,7 +315,7 @@ terminationGracePeriodSeconds: 10
 service:
   annotations: {}
-  clusterIP: None
+  # clusterIP:
```
I would leave `clusterIP`,  without the comment
Or maybe better `clusterIP: null`

austinorth replied:
Reverting to the original value after @steven-sheehy 's comment

Gsantomaggio commented on stable/rabbitmq-ha/Chart.yaml:
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 appVersion: 3.8.0
-version: 1.36.0
+version: 1.37.0
```
`version` should be `1.36.1` since is a minor fix

steven-sheehy reviewed (changes requested):
This is a major breaking change. `.spec.clusterIP` is an immutable field and this changes the default from `None` to "", breaking current deployments that use the default value of `None`. The upgrade would fail and they'd have to manually delete the service for it to succeed. I'm inclined not to merge this.

If you'd like to specifically fix the referenced issue by disallowing `None` when type LoadBalancer, I would consider that.

Commit:
Implemented changes suggested by reviewers

Reverted to the previous settings for the value and instead updated the
logic in the service template to avoid setting clusterIP value when type
is LoadBalancer and it is set to "None".

Signed-off-by: Austin Orth <aorth@niche.com>

Commit:
Merge branch 'master' into rabbitmq-ha

steven-sheehy reviewed (approved):
/lgtm
Thanks!
